mmcv/ops/chamfer_distance.py

```diff
@@ -0,0 +1,110 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import torch
+from torch import nn
+from torch.autograd import Function
+from torch.autograd.function import once_differentiable
+
+from ..utils import ext_loader
+
+ext_module = ext_loader.load_ext(
+    '_ext', ['chamfer_distance_forward', 'chamfer_distance_backward'])
+
+
+class ChamferDistanceFunction(Function):
+    """This is an implementation of the 2D Chamfer Distance.
+
+    It has been used in the paper `Oriented RepPoints for Aerial Object
+    Detection (CVPR 2022) <https://arxiv.org/abs/2105.11111>_`.
+    """
+
+    @staticmethod
+    def forward(ctx, xyz1, xyz2):
+        """
+        Args:
+            xyz1 (torch.Tensor): Point set with shape (B, N, 2).
+            xyz2 (torch.Tensor): Point set with shape (B, N, 2).
+
+        Returns:
+            tuple:
+
+                - dist1 (torch.Tensor): Chamfer ditacne (xyz1 to xyz2) with
+                    shape (B, N).
+                - dist2 (torch.Tensor): Chamfer ditacne (xyz2 to xyz1) with
+                    shape (B, N).
+                - idx1 (torch.Tensor): Index of chamfer ditacne (xyz1 to xyz2)
+                    with shape (B, N), which be used in compute gradient.
+                - idx2 (torch.Tensor): Index of chamfer ditacne (xyz2 to xyz2)
+                    with shape (B, N), which be used in compute gradient.
+        """
+        batchsize, n, _ = xyz1.size()
+        _, m, _ = xyz2.size()
+        device = xyz1.device
+
+        dist1 = torch.zeros(batchsize, n)
+        dist2 = torch.zeros(batchsize, m)
+
+        idx1 = torch.zeros(batchsize, n).type(torch.IntTensor)
+        idx2 = torch.zeros(batchsize, m).type(torch.IntTensor)
+
+        dist1 = dist1.to(device)
+        dist2 = dist2.to(device)
+        idx1 = idx1.to(device)
+        idx2 = idx2.to(device)
+        torch.cuda.set_device(device)
```

Comment:
Can we remove `set_device`? We already have device guard, which is an RAII `set_device`.

Reply:
ok